src/PAModel/Serializers/SourceSerializer.cs

```diff
@@ -795,5 +797,78 @@ private static BuildVerJson GetBuildDetails()
             }
         }
 
+        private static void ValidateIfTemplateExists(CanvasDocument app, BlockNode node, ErrorContainer errors)
```

Comment:
What is the purpose of this wrapper? it looks like the start of ValidateIfTemplateExistsInternal is basically the same?

Reply:
this wrapper takes in node from one of the source files, basically the parent root node like a screen. This wrapper is iterating through the children of the screen and the the other methods makes sure to iterate through all the other individual controls. 
I was trying to do it in one method recursively but found some instances where non group/groupContainer controls also had children.